test/color/test_gray_core.py

```diff
@@ -0,0 +1,74 @@
+import os
+
+import jax
+import numpy as np
+import pytest
+import torch
+
+from kornia.color.gray_core import grayscale_to_rgb
+from kornia.core import Device, Dtype
+from kornia.image.base import ChannelsOrder, ColorSpace, ImageLayout, ImageSize, PixelFormat
+from kornia.image.image import Image
+from kornia.testing import BaseTester
+
+# TODO: this should be done in the conftest.py
+os.environ["KERAS_BACKEND"] = "torch"
+
+
+def _make_random_image_u8(shape, device, dtype, backend):
+    if backend == "torch":
+        return torch.randint(0, 255, shape, device=device)
+    elif backend == "jax":
+        return jax.random.randint(jax.random.PRNGKey(0), shape=shape, minval=0, maxval=255)
+    elif backend == "numpy":
+        rng = np.random.default_rng(0)
+        return rng.integers(0, 255, shape, dtype=np.uint8)
+    else:
+        raise NotImplementedError(backend)
+
+
+class TestGrayCore(BaseTester):
+    # @pytest.mark.parametrize("backend", ["torch"])
+    # @pytest.mark.parametrize("backend", ["numpy"])
+    @pytest.mark.parametrize("backend", ["jax"])
+    @pytest.mark.parametrize("shape", [(4, 5), (5, 4)])
+    @pytest.mark.parametrize("num_channels", [1])
+    def test_smoke(self, backend, device, dtype, num_channels, shape):
+        # NOTE: just to make sure that the function is working
+        if backend in ["torch", "jax"]:
+            _shape = (num_channels, *shape)
+            channels_order = ChannelsOrder.CHANNELS_FIRST
+        else:
+            _shape = (*shape, num_channels)
+            channels_order = ChannelsOrder.CHANNELS_LAST
+        image_gray_data = _make_random_image_u8(_shape, device, dtype, backend)
+        image_gray = Image(
+            data=image_gray_data,
+            pixel_format=PixelFormat(color_space=ColorSpace.GRAY, bit_depth=8),
+            layout=ImageLayout(
```

Comment:
this is looking too complex for an API to replace a simple tensor -- maybe we can pass the `channels_order` into the Image constructor, and from this construct what the layout is

Reply:
Also for keras core we should consider using the `keras_core.config.image_data_format()` as the default value for the layout

https://github.com/keras-team/keras-core/blob/v0.1.3/keras_core/backend/config.py#L106

Reply:
Also, use this flag to generate the inputs under the test suite, setting it like the actual device/dtype fixture (https://keras.io/keras_core/api/utils/config_utils/#setimagedataformat-function)

Reply:
sure we iterate that